docs/developer/cli-arguments.md

```diff
@@ -88,5 +90,83 @@ Flags:
 
 Use "kube-state-metrics [command] --help" for more information about a command.
 ```
+
+## Config file format
+
+The above CLI options may also be set via a YAML-format config-file whose path is supplied as the `--config` argument on the CLI.
+
+The option names are defined in [`type Options` in `pkg/options/options.go`](https://github.com/kubernetes/kube-state-metrics/blob/main/pkg/options/options.go#L42).
+
+A sample configuration might look like:
+
+```yaml
+    auto-gomemlimit: true
+    auto-gomemlimit-ratio: 0.9
+    # Equivalent of --metric-labels-allowlist CLI option. type LabelsAllowList map[string][]string
+    labels_allow_list:
+      "*":
+        # will map to {resourcekind}_labels label label_k8s_kapp_k14s_io_app:
+        - kapp.k14s.io/app
+        - mycompany/some-label
+        - mycompany/anotherlabel
+      # check kube_node_info before adding info here:
+      nodes:
+        - kubernetes.io/os
+        - kubernetes.io/arch
+        - node.kubernetes.io/instance-type
+        - eks.amazonaws.com/nodegroup
+        - eks.amazonaws.com/nodegroup-image
+```
+
+### Using a config file with Kubernetes
```

Comment:
This isn't specific to KSM, but general knowledge to mount any configmap. Can we drop this?